src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -156,14 +157,21 @@ void OTAImageProcessorImpl::HandleProcessBlock(intptr_t context)
         ChipLogError(SoftwareUpdate, "mDownloader is null");
         return;
     }
-    esp_err_t err = esp_ota_write(imageProcessor->mOTAUpdateHandle, imageProcessor->mBlock.data(), imageProcessor->mBlock.size());
-    if (err != ESP_OK)
+
+    ByteSpan block = ByteSpan(imageProcessor->mBlock.data(), imageProcessor->mBlock.size());
+    if (imageProcessor->ProcessHeader(block) == CHIP_NO_ERROR)
```

Comment:
You should probably call `EndDownload` when the header parsing fails to gracefully close the BDX session.

Reply:
Thanks for pointing it out. Ending download and returning error.